datajunction-server/tests/api/client_test.py

```diff
@@ -1,96 +1,146 @@
 """
 Tests for client code generator.
 """
+import os
+from pathlib import Path
+
 import pytest
 from httpx import AsyncClient
 
+TEST_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+@pytest.fixture
+def load_expected_file():
+    """
+    Loads expected fixture file
+    """
+
+    def _load(filename: str):
+        expected_path = TEST_DIR / Path("files/client_test")
+        with open(expected_path / filename, encoding="utf-8") as fe:
+            return fe.read().strip()
+
+    return _load
+
+
+def trim_trailing_whitespace(string: str) -> str:
+    """Trim trailing whitespace on each line"""
+    return "\n".join([chunk.rstrip() for chunk in string.split("\n")]).strip()
+
 
 @pytest.mark.asyncio
-async def test_generated_python_client_code_new_metric(client_with_roads: AsyncClient):
+async def test_generated_python_client_code_new_source(
+    module__client_with_roads: AsyncClient,
```

Comment:
@agorajek I switched all of these to use the `module__*` AsyncClients rather than the default ones. This one's fairly easy since the entire test should be read-only

Reply:
Never mind, I'm going to remove this from this PR in favor of the one that does this for all tests